### PR TITLE
tests(geometry): migrate geometry module to use BaseTester.

### PR DESCRIPTION
**Pull request by Big-Marvel**
This PR migrates `tests/geometry` module to use `BaseTester` as part of the ongoing test standardization effort.

**Fixes/Relates to:** #2752 

## 🛠️ Changes Made
- [x] Migrated test classes of tests\geometry module to inherit from BaseTester for test standardization.

---

## 🧪 How Was This Tested?
- [x] **Unit Tests:** (List new/updated tests)
- [ ] **Manual Verification:** (Describe the steps you took)
- [ ] **Performance/Edge Cases:** (How does this handle nulls, large data, etc.?)

---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [x] 🟢 **No AI used.**
- [ ] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [ ] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [x] I am assigned to the linked issue (required before PR submission)
- [x] The linked issue has been approved by a maintainer
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] (Optional) I have attached screenshots/recordings for UI changes.

### CODE REVIEW

File: tests/geometry/transform/test_thin_plate_spline.py [hunk 1]
```diff
@@ -227,11 +228,16 @@ def test_exception(self, batch_size, device, dtype):
     @pytest.mark.grad()
     @pytest.mark.parametrize("batch_size", [1, 3])
     def test_gradcheck(self, batch_size, device, dtype):
+        if device.type != "cpu":
+            pytest.skip("gradcheck is unstable for warp_image_tps on CUDA")
```
Comment (edgarriba) [inline tests/geometry/transform/test_thin_plate_spline.py:232]: can we investigate why, maybe open a ticket

**Review by edgarriba — APPROVED** [review]
LGTM, make ci happy

**Comment by Big-Marvel** [comment]
Hi @edgarriba,
The grad check failure happens specifically for `warp_image_tps`, which internally uses
`torch.nn.functional.grid_sample.`
Since `grid_sample` has piecewise / interpolation-based gradients, finite-difference `gradcheck` seems numerically unstable here.

would you suggest 
- Skipping `gradcheck` for `warp_image_tps` entirely
- Passing dst as the centers for the radial basis functions,
currently the TPS formula is being evaluated using weights calculated for a completely different set of coordinates.
- Use a smooth image (like a linear ramp) for gradcheck.

Also , test are passing in the local environment but not in githubCI.

File: tests/geometry/transform/test_thin_plate_spline.py (hunk 1 above)
Comment (Big-Marvel) [reply tests/geometry/transform/test_thin_plate_spline.py:232]: I looked into this while debugging the `gradcheck` failures.
`warp_image_tps` internally relies on `torch.nn.functional.grid_sample`. On CUDA,
`grid_sample` uses different kernels and non-deterministic atomic paths, and its
gradients are known to be numerically unstable under finite-difference `gradcheck`.

I agree it would be good to document this properly. I can open a ticket to track CUDA `gradcheck` instability for `warp_image_tps` (and more generally `grid_sample-based ops).

**Comment by Big-Marvel** [comment]
The persistent mismatch happens for input 2 (= TPS kernel params). 
Even on CPU+float64 and larger image size (changed it to 32x32), finite-difference gradcheck is unstable wrt TPS parameters because the warp depends on grid_sample and small parameter perturbations change interpolation neighborhoods (piecewise gradients).

I'd suggest to keep gradcheck for warp_image_tps but restrict it to checking gradients wrt image only (treat kernel/affine as constants), since that’s stable and still validates backward correctness.

let me know your thoughts on this @edgarriba

File: tests/geometry/transform/test_thin_plate_spline.py (hunk 1 above)
Comment (edgarriba) [reply tests/geometry/transform/test_thin_plate_spline.py:232]: sounds good. We might want also to think how to not skip this kind of test but instead just collect properly during the pytest call

Comment (Big-Marvel) [reply tests/geometry/transform/test_thin_plate_spline.py:232]: I agree. Since this PR is already focused on migrating tests to `BaseTester`, I’ll keep it scoped and open a follow-up PR to handle pytest-side collection/config (so gradcheck tests run only on CPU jobs rather than skipping inside the test).
